libraries/AP_MSP/AP_MSP_Telem_DJI.cpp

```diff
@@ -104,4 +104,31 @@ MSPCommandResult AP_MSP_Telem_DJI::msp_process_out_fc_variant(sbuf_t *dst)
     return MSP_RESULT_ACK;
 }
 
+MSPCommandResult AP_MSP_Telem_DJI::msp_process_out_esc_sensor_data(sbuf_t *dst)
+{
+#if HAL_WITH_ESC_TELEM
+    if (AP::msp()->_options & AP_MSP::OPTION_TELEMETRY_DJI_WORKAROUNDS) {
+        AP_ESC_Telem& telem = AP::esc_telem();
+        if (telem.get_last_telem_data_ms(0)) {
+            int16_t max_temperature = INT16_MIN;
+            float rpm_acc = 0;
+            const uint8_t num_motors = telem.get_num_active_escs();
+            for (uint8_t i = 0; i < num_motors; i++) {
+                int16_t temp = 0;
+                float rpm = 0.0f;
+                telem.get_rpm(i, rpm);
```

Comment:
There is already a function to compute the average rpm - you should use it. You are not checking the return value here, the rpm returned could be out of range

Reply:
I assumed the code in `AP_MSP_Telem_Backend.cpp` was good. This part is just a copy. But I'm going to take a look

Reply:
There is `get_average_motor_frequency_hz()` but I don't see a method to get the average RPM

Reply:
x60?

Reply:
I don't see any problem not checking the return value of `get_rpm()` or `get_temperature()` since the value of the reference won't change in case where it is not possible to get a value and it returns false. It is not like we could make the OSD element blink with DJI FPV to indicate a problem.

Perhaps you will like this new version better anyway

Reply:
> x60?

Yeah ok but I don't like that much. Using RPM to compute the frequency then using the frequency to compute back the RPM